src/pages/cart/useUpdateCart.tsx

```diff
@@ -0,0 +1,12 @@
+import { useMutation, UseMutationOptions } from "@tanstack/react-query";
+import { CartItem } from "../../utils/types";
+import { updateCartItem } from "../../utils/routes";
+
+export function useUpdateCart(item: CartItem, options?: UseMutationOptions) {
```

Comment:
Might be nice to swap this options type out like we did for `useDeleteItem` to play more nicely with the `react-query` types.